File: jiaoyu-teacher-web/src/main/webapp/js/service/secondmenuService.js
//服务层
app.service('secondmenuService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../secondmenu/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../secondmenu/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../secondmenu/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../secondmenu/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../secondmenu/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../secondmenu/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../secondmenu/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: jiaoyu-teacher-interface/src/main/java/com/jiaoyu/teacher/service/SecondmenuService.java
package com.jiaoyu.teacher.service;
import java.util.List;
import com.jiaoyu.pojo.Secondmenu;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SecondmenuService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Secondmenu> findAll();
	
	/**
	 * 返回父id的全部子集
	 * @param id
	 * @return
	 */
	public List<Secondmenu> findByParentId(Integer id);
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Secondmenu secondmenu);
	
	
	/**
	 * 修改
	 */
	public void update(Secondmenu secondmenu);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Secondmenu findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Secondmenu secondmenu, int pageNum,int pageSize);
	
}
